internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -371,17 +376,15 @@ object Incremental {
             s"early output must match -Ypickle-write path '$p' but was '$earlyJar' instead"
           )
       }
-    } else if (!isPickleWrite && options.pipelining) {
-      log.warn(s"-Ypickle-write should be included into scalacOptions if pipelining is enabled")
```

Comment:
What happened to this warning?

Reply:
I realized that this warning doesn't make sense because the fact that subproject A uses pipelining and A exports early output are actually orthogonal. I can start the compilation early but tell the downstream to wait till bytecode is available.

Reply:
If the condition that needs to checked for coherence is now `earlyOutput.isDefined` instead of `options.pipelining` then this should've switched to that instead of just being deleted.

Reply:
Yea. If `isPickleWrite` and `earlyJar` did not match, it would error out in the code above.

Reply:
If `-Ypickle-write` is missing and (export) pipelining was enabled it used to warn `"-Ypickle-write should be included into scalacOptions if pipelining is enabled"` while now it warns `"-Ypickle-write is specified but <path> is not?"`.

Reply:
ok. I guess we need a warning for either `None` cases, and fix the `case (None, _)` to `case (_, None)`?

Reply:
If you want you can juggle that.  To me it would seem more straightforward to just revert and tweak `s/options.pipelining/earlyOutput.isDefined/` here.